core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -55,141 +50,56 @@
-   * @param blockSize the block size
-   * @param blockStart the position at which the block starts in the file
-   * @param mountId the id of the UFS which the mount of this file is mapped to
-   * @param workerNetAddress the worker network address
-   * @param options the options
-   * @return the {@link BlockInStream} created
+   * @param blockSize the block size in bytes
+   * @param address the Alluxio worker address
+   * @param openUfsBlockOptions the options to open a UFS block, set to null if this is block is
+   *        not persisted in UFS
+   * @param options the in stream options
+   * @return the {@link InputStream} object
    */
-  // TODO(peis): Use options idiom (ALLUXIO-2579).
-  public static BlockInStream createUfsBlockInStream(FileSystemContext context, String ufsPath,
-      long blockId, long blockSize, long blockStart, long mountId,
-      WorkerNetAddress workerNetAddress, InStreamOptions options) throws IOException {
-    Closer closer = Closer.create();
-    try {
-      BlockWorkerClient blockWorkerClient =
-          closer.register(context.createBlockWorkerClient(workerNetAddress));
-      LockBlockOptions lockBlockOptions =
-          LockBlockOptions.defaults().setUfsPath(ufsPath).setOffset(blockStart)
-              .setBlockSize(blockSize).setMaxUfsReadConcurrency(options.getMaxUfsReadConcurrency())
-              .setMountId(mountId);
-
-      LockBlockResult lockBlockResult =
-          closer.register(blockWorkerClient.lockUfsBlock(blockId, lockBlockOptions)).getResult();
-      PacketInStream inStream;
-      if (lockBlockResult.getLockBlockStatus().blockInAlluxio()) {
-        boolean local = workerNetAddress.getHost().equals(NetworkAddressUtils.getClientHostName());
-        if (local && Configuration.getBoolean(PropertyKey.USER_SHORT_CIRCUIT_ENABLED)
-            && !NettyUtils.isDomainSocketSupported(workerNetAddress)) {
-          inStream = closer.register(PacketInStream
-              .createLocalPacketInStream(
-                  lockBlockResult.getBlockPath(), blockId, blockSize, options));
-        } else {
-          inStream = closer.register(PacketInStream
-              .createNettyPacketInStream(context, workerNetAddress, blockId,
-                  lockBlockResult.getLockId(), blockWorkerClient.getSessionId(), blockSize, false,
-                  Protocol.RequestType.ALLUXIO_BLOCK, options));
-        }
-        blockWorkerClient.accessBlock(blockId);
-      } else {
-        Preconditions.checkState(lockBlockResult.getLockBlockStatus().ufsTokenAcquired());
-        inStream = closer.register(PacketInStream
-            .createNettyPacketInStream(context, workerNetAddress, blockId,
-                lockBlockResult.getLockId(), blockWorkerClient.getSessionId(), blockSize,
-                !options.getAlluxioStorageType().isStore(), Protocol.RequestType.UFS_BLOCK,
-                options));
+  public static BlockInStream createBlockInStream(FileSystemContext context, long blockId,
```

Comment:
NIT: `Foo.createFoo` seems verbose as `Foo` is repeated twice. I would suggest this to be simply `create`. Same for `BlockOutStream.createBlockOutStream` to be `BlockOutStream.create`